src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -251,17 +272,65 @@ public void AddTXTRecord(string key, string value)
                 Log.Error(Globals.LogTag, "Failed to add the TXT record, Error: " + (DnssdError)ret);
                 NsdErrorFactory.ThrowDnssdException(ret);
             }
-
             byte[] txtValue;
-            ushort txtLength;
-            GetTxtRecord(out txtLength, out txtValue);
+            txtValue = GetTXTRecords();
+            ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, (ushort)txtValue.Length, txtValue);
+            if (ret != (int)DnssdError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to set the DNS resource record, Error: " + (DnssdError)ret);
+                NsdErrorFactory.ThrowDnssdException(ret);
+            }
+        }
 
-            ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, txtLength, txtValue);
+        /// <summary>
+        /// Set the record.
+        /// </summary>
+        /// <remarks>
+        /// TXT record should be added after registering the local service using RegisterService().
+        /// </remarks>
+        /// <returns>on success returns zero else returns error number.</returns>
+        /// <since_tizen> 9 </since_tizen>
+        /// <param name="type">type of the record to set</param>
```

Comment:
type of -> Type of, what are the types, can a user pass any ushort value?

Reply:
SetRecord() is removed